napari/benchmarks/benchmark_shapes_layer.py

```diff
@@ -27,48 +27,140 @@
     from napari.benchmarks.utils import Skip
 
 
-class Shapes2DSuite:
+class BackendType(Enum):
+    bermuda = (
+        auto()
+    )  # data processing using PartSegCore, ready to switch to bermuda
+    triangle = auto()  # data preprocessing using numba, edge triangulation using numba, triangulation using triangle
+    numba = auto()  # data preprocessing and edge triangulation using numba, triangulation using vispy
+    pure_python = auto()  # data preprocessing, edge triangulation, and triangulation using python and vispy
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        return self.name
+
+
+backend_list = list(BackendType)
+
+
+class _BackendSelection:
+    triangulate: Callable | None
+    prev_settings: bool
+    prev_numba: dict[str, Callable]
+
+    def _disable_numba(self):
+        try:
+            from napari.layers.shapes import (
+                _accelerated_triangulate_dispatch as atd,
+            )
+        except ImportError:
+            return
+
+        for name in atd.__all__:
+            if hasattr(atd, f'{name}_py'):
+                self.prev_numba[name] = getattr(atd, name)
+                setattr(atd, name, getattr(atd, f'{name}_py'))
```

Comment:
this is hard to get my head around. I'd consider adding some comments above. (And I agree with Carol's comment, let's maybe rename `_accelerated_triangulate_dispatch` to `triangle_dispatch` and use the complete import.

Reply:
added comment, could you check?